### DISCUSSION TITLE
Prisma schema doesn't match db schema

### DISCUSSION THREAD

**Discussion by erasebegin** [prisma/prisma#16040] [Q&A] [answered]
I've got a Person table that looks like this in the schema:
![image](https://user-images.githubusercontent.com/1922075/198577608-03aacdda-d303-47eb-9bf3-6202f6919282.png)
And a Role table that looks like this:
![image](https://user-images.githubusercontent.com/1922075/198577758-b4362fb9-2811-4d69-a4f0-10bc13ad796a.png)
But I can't see the `roles` column on `Person` in my database (Supabase psql). Neither can I see the relationship/join table that should have been created between `Person` and `Role`. So when I try to create a person with a role like so:
```javascript
      const person = await prisma.person.create({
        data: { ...req.body, roles: { createMany: { data: roles } } },
      });
```
I get the following error:
```
{
    data: {
      firstName: 'Person',
      lastName: 'Personicus',
      dob: new Date('2022-10-14T00:00:00.000Z'),
      gender: 'm',
      nationality: 'Krishlish',
      roles: {
        create: {
          id: '3b4f6d5f-68e9-4f14-9f83-1433d69d0fbc',
          name: 'vater',
          icon: 'person',
          teamId: '7b3cf5c0-1b64-4071-84dd-b7cbd2e3604d',
          personId: null
          ~~~~~~~~
        }
      }
    }
  }
  Unknown arg `personId` in data.roles.createMany.data.0.personId for type RoleCreateManyPersonInput. Did you mean `teamId`? Available args:
  type RoleCreateManyPersonInput {
    id?: String
    name: String
    icon?: String | Null
    teamId?: String | Null
  }

```
I've tried doing `prisma generate` to update the client, and I've done `prisma migrate` and `prisma db push` several times, but it always says "nothing to update."

Very confused, please help! 🙏

**Comment by austincrim** [comment, marked as answer]
Hey @erasebegin 👋,

The error you are receiving is actually because [you can't create relations inside of a `createMany` call](https://github.com/prisma/prisma/issues/5455). Your database structure is most likely correct (you won't see the `role` field inside of Postgres because that's a Prisma-only abstraction, and there won't be a join table because this is a one-to-many relation).

Feel free to read through the linked issue for workarounds and add a 👍 if you want to see us prioritize it.

**Reply by erasebegin** [reply]
Thank you so much for this illuminating answer. I think what I was actually trying to do was to do a nested `connect` rather than a nested `create`. But there were several other things I didn't realise which you have revealed to me, thank you 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
